src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -781,13 +781,20 @@ namespace smt::noodler {
         expr_ref lengths(m);
         std::unordered_set<BasicTerm> init_length_sensitive_vars{ get_init_length_vars(aut_assignment) };
 
-        DecisionProcedure dec_proc = DecisionProcedure{ instance, aut_assignment, init_length_sensitive_vars, m, m_util_s, m_util_a };
+        // use underapproximation to solve
+        if(m_params.m_underapproximation && solve_underapprox(instance, aut_assignment, init_length_sensitive_vars) == l_true) {
+            STRACE("str", tout << "underapprox sat \n";);
+            return FC_DONE;
+        }
+
```

Comment:
Better separate underapproximation from the regular algorithm checking the whole instance.
```suggestion
        // Underapproximation is unsat. We need to check the whole instance instead.
```

Reply:
Great, can you push the comment there?

Reply:
(Still remains to be added.)